src/ast/IntrinsicFunctor.h

```diff
@@ -61,37 +62,23 @@ class IntrinsicFunctor : public Functor {
     }
 
     /** Get function information */
-    const IntrinsicFunctorInfo* getFunctionInfo() const {
-        return info;
+    std::optional<FunctorOp> getFunctionOp() const {
+        return op;
     }
 
     /** Set function information */
-    void setFunctionInfo(const IntrinsicFunctorInfo& info) {
-        this->info = &info;
-    }
-
-    /** Get the return type of the functor. */
-    TypeAttribute getReturnType() const override {
-        assert(info && "functor info not yet available");
-        return info->result;
-    }
-
-    /** Get type of the functor argument*/
-    TypeAttribute getArgType(const size_t arg) const override {
-        assert(info && "functor info not yet available");
-        return info->params.at(info->variadic ? 0 : arg);
+    void setFunctionOp(FunctorOp op) {
+        this->op = op;
     }
 
     IntrinsicFunctor* clone() const override {
-        return new IntrinsicFunctor(function, info, souffle::clone(args), getSrcLoc());
+        return new IntrinsicFunctor(function, op, souffle::clone(args), getSrcLoc());
     }
 
 protected:
-    IntrinsicFunctor(
-            std::string op, const IntrinsicFunctorInfo* info, VecOwn<Argument> args, SrcLocation loc = {})
-            : Functor(std::move(args), std::move(loc)), function(std::move(op)), info(info) {
-        assert((!info || info->symbol == function) && "functor info must match symbol");
-    }
+    IntrinsicFunctor(std::string function, const std::optional<FunctorOp> op, VecOwn<Argument> args,
+            SrcLocation loc = {})
+            : Functor(std::move(args), std::move(loc)), function(std::move(function)), op(std::move(op)) {}
```

Comment:
```suggestion
    IntrinsicFunctor(std::string function, std::optional<FunctorOp> op, VecOwn<Argument> args,
            SrcLocation loc = {})
            : Functor(std::move(args), std::move(loc)), function(std::move(function)), op(op) {}
```
op is const so the move has no effect. It's also trivially copyable, which would also stop std::move having an effect